subprojects/kotlin-dsl-integ-tests/src/integTest/kotlin/org/gradle/kotlin/dsl/integration/KotlinDslJvmTargetIntegrationTest.kt

```diff
@@ -92,10 +92,7 @@ class KotlinDslJvmTargetIntegrationTest : AbstractPluginIntegrationTest() {
     }
 
     @Test
-    fun `can use java toolchain to compile precompiled scripts`() {
-
-        // https://github.com/gradle/gradle/issues/23125
-        assumeJava11OrHigher()
+    fun `can use Java Toolchain to compile precompiled scripts`() {
```

Comment:
This test now fails when running on Java 8 because it tries to use the plugin, built using a Java 11 toolchain, in the build.

This test should now be revisited to do the following:

Split the build in two:
* one to build the plugin
* one to use the plugin

Build the plugin running Gradle with the current test java version but using Java 11 toolchain.
Run the build that uses the plugin with Java 11.